dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -444,4 +480,19 @@ private static async Task<T> RunRequestAsync<T>(Func<Task<T>> request)
                 $"Something went wrong: {e.Message}", e);
         }
     }
+
+    /// <summary>
+    /// Captures usage details, including token information.
+    /// </summary>
+    /// <param name="usage">Instance of <see cref="CompletionsUsage"/> with usage details.</param>
+    private void CaptureUsageDetails(CompletionsUsage usage)
+    {
+        this.Logger.LogInformation(
+            "Prompt tokens: {PromptTokens}. Completion tokens: {CompletionTokens}. Total tokens: {TotalTokens}.",
+            usage.PromptTokens, usage.CompletionTokens, usage.TotalTokens);
+
+        s_promptTokensCounter.Add(usage.PromptTokens);
```

Comment:
Do we get correlationIds for this telemetry at all so we can know which function execution event corresponds to a usage event?

Reply:
Sure, in AppInsights console app example we output correlation id, so you can easily find each run by ID:
https://github.com/microsoft/semantic-kernel/blob/25ee1d5b6b03cf2811ccf7efb5eb4bcfe68c58c5/dotnet/samples/ApplicationInsightsExample/Program.cs#L58-L59

In web apps this field should be automatically populated with each API request.

Reply:
Okay I'll check this out and see if it matches my expectations. Thanks!